data/ai/micro_ais/micro_ai_helper.lua

```diff
@@ -3,6 +3,56 @@ local MAIUV = wesnoth.require "ai/micro_ais/micro_ai_unit_variables.lua"
 
 local micro_ai_helper = {}
 
+local function is_number(value, check_int, ai_type, name)
+    -- Allow any number for 'float' type.
+    -- Allow anything that has an integer representation for 'integer', but
+    -- issue warning if it is given in 'float' notation.
+    local number = tonumber(value)
+    if (not number) then
+        return false
+    elseif check_int then
+        if (number ~= math.floor(number)) then
+            return false
+        elseif (math.type(number) ~= 'integer') then
+            std_print("Warning: [micro_ai] tag (" .. ai_type .. ") parameter '" .. name .. "' must be an integer. It has an integer representation, but is provided in floating-point format.")
+        end
+    end
+
+    return true
+end
+
+local function check_key_type(k_name, k_value, k_type, ai_type)
+    if k_value then
+        local is_wrong_type = false
+        if (k_type == 'float') then
+            is_wrong_type = not is_number(k_value, false, ai_type, k_name)
+        elseif (k_type == 'integer') then
+            is_wrong_type = not is_number(k_value, true, ai_type, k_name)
+        elseif (k_type == 'csv_integer') then
```

Comment:
1. I'm not entirely sold on the name here. I wonder if "int_list" or "integer_list" would be easier to understand?

2. Maybe this is a low priority, but is there any demand for a list of real numbers?